### PR TITLE
feat: add baln staking apy and update pool info

### PR DESCRIPTION
**Pull request by hetfly**

### CODE REVIEW

**Comment by BennyOptions** [comment]
How to calculate:

($$ value of fees distributed over the last month * 12) / (Number of Staked BALN * $$ price of BALN)
Screen Shot 2022-01-03 at 11 42 42 PM


Screen Shot 2022-01-03 at 11 44 55 PM
Screen Shot 2022-01-03 at 11 44 36 PM

**Comment by BennyOptions** [comment]
For Fee APYs, here is the calculation for **everything except sICX/ICX**. For sICX / ICX, instead of multiplying by 50%, please multiply by 70%

$$ value of fees * 365 * 50% / $$ value of liquidity.... (Red box * 365 * 50% / Green Box)
Screen Shot 2022-01-03 at 11 49 24 PM

**Comment by hetfly** [comment]
@BennyOptions  thanks for the help! Please review the numbers.

**Comment by BennyOptions** [comment]
@parrot9design @hetfly I just realized that the calculation I gave for Fee APY for pools is only based on the last 24 hours, not on a trailing 30 day average.

@hetfly @pranav925 is it possible to pull Fees (30 day)? If so, let's change the Fee APY calculation to use Fees (30d) * 12 instead of Fee (24h) * 365.

**Comment by BennyOptions** [comment]
For 30d of fees

https://balanced.geometry.io/api/v1/stats/dex-pool-stats-30d

**Comment by hetfly** [comment]
updated, ready for review

File: src/sections/PairSection.tsx [hunk 1]
```diff
@@ -67,8 +81,24 @@ export default function PairSection() {
...
+            <HeaderText minWidth={'135px'}>
+              <Typography marginRight={'5px'}>APY</Typography>
+              <MouseoverTooltip
+                width={330}
+                text={
+                  <>
+                    The BALN APY is calculated from the USD value of BALN rewards available for a pool.
+                    <br />
+                    <br />
+                    The fee APY is calculated from the swap fees added to a pool in the last 30 days.
```
Comment (parrot9design) [inline src/sections/PairSection.tsx:94]:
```suggestion
                    The fee APY is calculated from the swap fees earned by a pool in the last 30 days.
```

**Comment by parrot9design** [comment]
This is looking great, thanks @hetfly! Aside from the minor content change suggested, I only noticed 2 things:

- [ ] The colour for the day should be `rgba(255,255,255,0.75)`

- [ ] Let's move the Exchange APY label to the left a bit, so the `?` is aligned with the data underneath. It doesn't look as bad on desktop, but when you're on mobile you likely have to scroll horizontally before you even see that there's a tooltip beside APY:

Screen Shot 2022-01-18 at 9 27 02 AM

**Review by parrot9design — APPROVED** [review]
Looks good to me, thanks @hetfly!

**Comment by BennyOptions** [comment]
Blocked, waiting for continuous rewards upgrade

**Comment by BennyOptions** [comment]
@hetfly I was just comparing the APYs from this PR to production and noticed they are quite different. Can you explore that? It's the BALN APY. They should both match.

This PR
Screen Shot 2022-02-22 at 6 56 48 PM


Current production page
Screen Shot 2022-02-22 at 6 58 04 PM

**Comment by hetfly** [comment]
@BennyOptions APY calculations are identical, difference was in current rewards settings. Merged master into this branch and all looks good now.

**Comment by parrot9design** [comment]
@hetfly Let's resolve the conflicts in this branch so we can have it ready to merge alongside the continuous rewards update.

**Comment by hetfly** [comment]
@parrot9design this PR is ready to be reviewed. 

BTW shouldn't we change the abbreviations to APR (BALN staking as well as pools)? APY reflects compounding during the year, our numbers don't.
The BALN APY is something I'd like to go through in Stats v2 

cc @BennyOptions

File: src/pages/StatsPage/index.tsx [hunk 2]
```diff
@@ -185,24 +193,40 @@ export function StatsPage() {
...
+                          ).
+                        </strong>
+                      </>
+                    }
+                    placement="top"
+                  >
+                    <QuestionIcon width={14} />
+                  </MouseoverTooltip>
+                </Flex>
+                <Typography>BALN staking APY</Typography>
```
Comment (parrot9design) [inline src/pages/StatsPage/index.tsx:229]:
```suggestion
                <Typography>BALN staking APR</Typography>
```

File: src/sections/PairSection.tsx [hunk 3]
```diff
@@ -165,14 +206,31 @@ export default function PairSection() {
...
-                  key: 'apy',
+                  key: 'apyTotal',
                 })
               }
             >
+              <MouseoverTooltip
+                width={330}
+                text={
+                  <>
+                    The BALN APY is calculated from the USD value of BALN rewards available for a pool.
```
Comment (parrot9design) [inline src/sections/PairSection.tsx:222]:
```suggestion
                    The BALN APR is calculated from the USD value of BALN rewards available for a pool.
```

File: src/sections/PairSection.tsx [hunk 4]
```diff
@@ -165,14 +206,31 @@ export default function PairSection() {
...
+                    <br />
+                    <br />
+                    The fee APY is calculated from the swap fees earned by a pool in the last 30 days.
```
Comment (parrot9design) [inline src/sections/PairSection.tsx:225]:
```suggestion
                    The fee APR is calculated from the swap fees earned by a pool in the last 30 days.
```

File: src/sections/PairSection.tsx [hunk 5]
```diff
@@ -165,14 +206,31 @@ export default function PairSection() {
...
+                  </>
+                }
+                placement="top"
+              >
+                <QuestionWrapper>
+                  <QuestionIcon className="header-tooltip" width={14} />
+                </QuestionWrapper>
+              </MouseoverTooltip>
               APY
```
Comment (parrot9design) [inline src/sections/PairSection.tsx:234]:
```suggestion
              APR
```

**Comment by parrot9design** [comment]
Ah, yes I think you're correct re: APY vs APR. I've added suggestions to make the change easier. Other than that, this should be good to merge alongside continuous rewards.

**Comment by parrot9design** [comment]
@hetfly I noticed that the BALN staking APY doesn't have a loading animation. Is it possible to add one?

The Exchange section currently isn't loading any data. Does that need to be updated to account for the new java contracts?

**Comment by parrot9design** [comment]
Looks good, thanks @hetfly!

**Comment by BennyOptions** [comment]
Math looks good, no longer blocked by contract. @hetfly @parrot9design looks good to merge

**Comment by BennyOptions** [comment]
Might be worth noting here that the calculation doesn't include impermanent losses, with a link to impermanent loss explained? 

https://finematics.com/impermanent-loss-explained/#:~:text=In%20essence%2C%20impermanent%20loss%20is,providing%20liquidity%20in%20that%20asset.

**Comment by parrot9design** [comment]
Good call. @hetfly, let's add this line to the end as a label:

> Impermanent loss is not factored in.
Screen Shot 2022-06-21 at 5 14 50 PM

If it's possible to keep the tooltip content up when the mouse hovers over it, we could link off for more info, like this:

> [Impermanent loss](https://finematics.com/impermanent-loss-explained/) is not factored in.

But that may be too much hassle, so it's also fine to ship without it.

**Pull request merged**